tasks/main.yml

```diff
@@ -1,4 +1,10 @@
 ---
+- name: Check Terraform Version
+  shell: terraform --version | grep {{ terraform_version }}
+  register: current_terraform_version
+  changed_when: current_terraform_version.rc != 0
```

Comment:
It isn't clear what this `changed_when` contributes to the task. Omitting it seems to produce the same behavior with the `unarchive` task.